compose/parallel.py

```diff
@@ -133,3 +92,86 @@ def parallel_kill(containers, options):
 
 def parallel_restart(containers, options):
     parallel_operation(containers, 'restart', options, 'Restarting')
+
+
+def _dep_ordered_do_operation(operation, obj, callback):
+    try:
+        result = operation(obj)
+        callback((obj, result, None))
+    except Exception as e:
+        callback((obj, None, e))
+
+
+def dep_ordered_op(objects, operation, get_deps, get_name, operation_name="",
+                   silent=False):
+    """Runs operation on objects in parallel while ensuring that operation is
+    ran on object only after it is ran on all its dependencies.
+
+    get_deps called on object must return a collection with its dependencies.
+    get_name called on object must return its name.
+    Setting silent to True disables any console output.
+    """
+    objects = list(objects)
+    stream = get_output_stream(sys.stderr)
+    writer_class = DummyParallelStreamWriter if silent else ParallelStreamWriter
```

Comment:
I don't think we need to worry about being silent. We can have start and up print a similar message. If they already have logging somewhere else we can move it here.